From 831194f0be7733c99c7a2c69d9e9695b82e05010 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Poho=C5=99elsk=C3=BD?= <opohorel@redhat.com>
Date: Thu, 30 Jan 2025 08:48:47 +0100
Subject: [PATCH] time: fix compiling with GCC15
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

GCC15 complains about incompatible pointer type in run_command()
Initialize interrupt_signal and quit_signal with correct type

Upstream-Status: Submitted [https://lists.gnu.org/archive/html/bug-time/2025-01/msg00000.html]
Signed-off-by: Ondřej Pohořelský <opohorel@redhat.com>
---
 src/time.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/time.c b/src/time.c
index 7b401bc..c8d7ab0 100644
--- a/src/time.c
+++ b/src/time.c
@@ -803,7 +803,7 @@ run_command (cmd, resp)
      RESUSE *resp;
 {
   pid_t pid;			/* Pid of child.  */
-  sighandler interrupt_signal, quit_signal;
+  __sighandler_t interrupt_signal, quit_signal;
   int saved_errno;
 
   resuse_start (resp);
-- 
2.48.1

